core/common/src/main/scala/org/typelevel/otel4s/Attributes.scala

```diff
@@ -93,43 +99,74 @@ sealed trait Attributes
     * [[scala.collection.IterableOps.concat `IterableOps#concat`]] that returns
     * `Attributes` rather than `Iterable`.
     *
-    * If multiple [[`Attribute`]]s in `this` and/or `that` have the same key,
+    * If multiple [[`Attribute`]]s in `this` and/or `that` have the same name,
     * only the final one (according to `that`'s iterator) will be retained in
     * the resulting `Attributes`.
     */
   def concat(that: IterableOnce[Attribute[_]]): Attributes =
-    attributesFactory.fromSpecific(this.view ++ that)
+    that match {
+      case other: Attributes =>
+        uncheckedFromMap(asUntypedMap ++ other.asUntypedMap)
+      case other =>
+        uncheckedFromMap(
+          asUntypedMap ++ other.iterator.map(a => a.key.name -> a)
+        )
+    }
 
   /** Invariant overload of [[scala.collection.IterableOps.++ `IterableOps#++`]]
     * that returns `Attributes` rather than `Iterable`.
     *
-    * If multiple [[`Attribute`]]s in `this` and/or `that` have the same key,
+    * If multiple [[`Attribute`]]s in `this` and/or `that` have the same name,
     * only the final one (according to `that`'s iterator) will be retained in
     * the resulting `Attributes`.
     */
   final def ++(that: IterableOnce[Attribute[_]]): Attributes =
     concat(that)
 
+  /** Removes all attributes with any of the given key names. */
+  def removedAllUntyped(names: IterableOnce[String]): Attributes =
+    uncheckedFromMap(asUntypedMap.removedAll(names))
+
   /** Removes all attributes with any of the given keys. */
-  def removedAll(that: IterableOnce[AttributeKey[_]]): Attributes =
-    attributesFactory.fromSpecific((toMap -- that).values)
+  def removedAll(keys: IterableOnce[AttributeKey[_]]): Attributes =
+    keys.iterator.foldLeft(this)(_ - _)
```

Comment:
```suggestion
    removedAllUntyped(keys.iterator.map(_.name))
```

Reply:
that's not actually the same—the current implementation only removes keys where the name *and type* match